build-system/tasks/check-types.js

```diff
@@ -35,7 +36,53 @@ const {typecheckNewServer} = require('../server/typescript-compile');
  * Note: This is a TEMPORARY holding point during the transition to type-safety.
  * @type {!Array<string>}
  */
-const PRIDE_FILES_GLOBS = ['src/resolved-promise.js', 'src/types.js'];
+const PRIDE_FILES_GLOBS = [
+  // Core
+  'src/core/**/*.js',
```

Comment:
```suggestion
  'src/core{,/**}/*.js',
```

Annoying quirk: `minimatch` (used by Node) treats `./**/*.js` as all files and files in subdirs; `globby` seems to only identify files in subdirs, but not files in the dir itself.

Reply:
Drive by comment: I'm pretrty sure this change is not necessary. Two pieces of evidence:

1. In https://github.com/ampproject/amphtml/pull/33957#discussion_r623201665, `src/core/**/*.js` appears to match `src/core/3p-frame.js`
2. Running `amp unit --headless --files=test/unit/**/test-log.js` will match `test/unit/test-log.js`

Also, the `fast-glob` [documentation](https://github.com/mrmlnc/fast-glob#basic-syntax) (used under the covers by `globby`) confirms this.

If you're observing that the correct set of files isn't being matched, could something else be at play?

Reply:
Lets pursue this question in a followup PR. I agree that something is fishy

Reply:
Agreed, confirmed locally though I distinctly recall fighting with this issue for a while, and now I have no idea from whence it came